internal/cmd/controller/gitops/reconciler/status_controller.go

```diff
@@ -12,19 +13,23 @@ import (
 	"github.com/rancher/fleet/pkg/durations"
 	"github.com/rancher/fleet/pkg/sharding"
 	"github.com/rancher/wrangler/v3/pkg/genericcondition"
-
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
 	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/builder"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
+// enqueueDelay is used as an artificial delay for enqueuing GitRepo status reconciliation requests
+// This allows aggregating multiple consecutive Bundle update events, reducing the number of GitRepo status changes at the cost of introducing a delay in the notification
+const enqueueDelay = 3 * time.Second
```

Comment:
Where does this value come from? I'm interested in the process behind determining it.
(not talking about making it configurable... yet ;) )

Reply:
I must admit I didn't follow a very scientific approach to pick this value, which should be big enough to defeat its purpose but not so much to affect user experience.
I considered adding some kind of adaptive mechanism but that increased the complexity for not much benefit (IMO).